### ISSUE TITLE
Chat Duplicates

### ISSUE THREAD

**Issue by ysiwnb** [ESX-Org/esx_rpchat#17]
Whenever i enable esx_rpchat, every command such as /ooc is duplicated in chat. any idea why?

**Comment by ElPumpo** [comment]
Because you have more than one handler for the command...

**Comment by x0Z3ro0x** [comment]
@ElPumpo what handler would that be in a fresh install? Having the same issue and stumbled upon this on Google. Took your advice and looked for a second resource handling those commands. Found none so took it a step further and used a fresh FXServer and STILL got the double chat.

For me this isn't an issue of a second resource handling the commands as there are none. YOURS is the only and the second chat being sent ISN'T OOC, TWT, or ME, it's plain white text that matches your statement in the previous command.

example: I use /twt Hello

everyone sees "(Twitter)Name: Hello" but they also see a second chat sent in plain white text with no prefix: "Hello"

**Comment by x0Z3ro0x** [comment]
After extensive testing, and butchering the default FiveM chat resource. This is most definitely an issue with YOUR script @ElPumpo 

```
RegisterServerEvent('chat:init')
RegisterServerEvent('chat:addTemplate')
RegisterServerEvent('chat:addMessage')
RegisterServerEvent('chat:addSuggestion')
RegisterServerEvent('chat:removeSuggestion')
RegisterServerEvent('_chat:messageEntered')
RegisterServerEvent('chat:clear')
RegisterServerEvent('__cfx_internal:commandFallback')



AddEventHandler('_chat:messageEntered', function(author, color, message)
    if not message or not author then
        return
    end

    -- TriggerEvent('chatMessage', source, author, message)

    if not WasEventCanceled() then
        -- TriggerClientEvent('chatMessage', -1, author,  { 255, 255, 255 }, message)
    end

    print(author .. ': ' .. message)
end)

AddEventHandler('__cfx_internal:commandFallback', function(command)
    local name = GetPlayerName(source)

    -- TriggerEvent('chatMessage', source, name, '/' .. command)

    if not WasEventCanceled() then
        -- TriggerClientEvent('chatMessage', -1, name, { 255, 255, 255 }, '/' .. command) 
    end

    CancelEvent()
end)

-- player join messages
-- AddEventHandler('playerConnecting', function()
    -- TriggerClientEvent('chatMessage', -1, '', { 255, 255, 255 }, '^2* ' .. GetPlayerName(source) .. ' joined.')
-- end)

-- AddEventHandler('playerDropped', function(reason)
    -- TriggerClientEvent('chatMessage', -1, '', { 255, 255, 255 }, '^2* ' .. GetPlayerName(source) ..' left (' .. reason .. ')')
-- end)

RegisterCommand('say', function(source, args, rawCommand)
    TriggerClientEvent('chatMessage', -1, (source == 0) and '^2[SERVER ADMIN]^7' or GetPlayerName(source), { 255, 255, 255 }, rawCommand:sub(5))
end)
```

The only other chat resource that handles the chat is YOURS. Since the code above is from the default FiveM and can no longer enter chat messages, that means YOUR resource is officially the only one handling the chat.

**Comment by nearbyplayer** [comment]
Hello, thank you for your extremely detailed post; however, your attitude was not needed. This ticket was **CLOSED** on July 12, 2018. This repo also hasn't been updated since approximately August 2018.

**If you'd like to submit a pull request with your fix please do so, and we'll gladly review it.**

**Comment by x0Z3ro0x** [comment]
Already created a pull request earlier perhaps I did something wrong? That being said, I didn't want to create a new ticket when this one has existing users that may or may not have fixed their issues.

https://github.com/x0Z3ro0x/esx_rpchat/tree/patch-1

All I did was comment out your print lines so the duplicate chat doesn't show anymore. Users can add to their server/main.lua at lines 26, 40, and 54.

Changing the lines from this:

`print(('%s: %s'):format(name, args))`

to this:

`--print(('%s: %s'):format(name, args))`

Here is the changes I made about 20 hours ago;

https://github.com/x0Z3ro0x/esx_rpchat/commit/b59dcd85eab712c393c97cf599a75c26661e7b5e
